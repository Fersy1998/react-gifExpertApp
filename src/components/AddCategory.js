import React from 'react';
import { useState } from 'react';
import PropTypes from 'prop-types'

const AddCategory = ({setCategories}) => {

  const [inputValue, setinputValue] = useState("");
  
  const onChangeHandled=(e)=>{
    setinputValue(e.target.value);
    //console.log("llamado a onChangeHandled");
  }
  const submitHandled=(e)=>{
    e.preventDefault();  
    if (inputValue.trim().length>1){
      setCategories(categorias=>[inputValue, ...categorias]);
      setinputValue('');
    }
  }
  return (
  
    <form onSubmit={submitHandled}>
      <input 
      placeholder="Agregar CategorÃ­a"
      value={inputValue}
      onChange={onChangeHandled}
      />
    </form>
  
  )
}

AddCategory.propTypes = {
  setCategories:PropTypes.func.isRequired,
}

export default AddCategory