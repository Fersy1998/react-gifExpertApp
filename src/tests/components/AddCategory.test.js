import { shallow } from "enzyme";
import React from "react";
import AddCategory from "../../components/AddCategory";
import '@testing-library/jest-dom';
describe('Pruebas en AddCategory.js', ()=>{
    
    const setCategory=()=>jest.fn();
    let wrapper=shallow(<AddCategory setCategories={setCategory}/>);
    
    beforeEach (()=>{
        jest.clearAllMocks();
        wrapper=shallow(<AddCategory setCategories={setCategory}/>);
    })
    test('should show perfectly', () => { 
        expect(wrapper).toMatchSnapshot();
    
    })
    test('should change the imput box', () => { 
        const input=wrapper.find('input');
        const value="Hamtaro"
        input.simulate('change', {
            target:{
            value:value
            }});
        }
    )
            
    test('shouldnt post information onsubmit', () => { 
        const form = wrapper.find("form");
        form.simulate('submit', {preventDefault(){}});
        expect(setCategory).not.toHaveBeenCalled;
    })
    
    test('should call the categories and clean the textbox', () => { 
        //Simulación del cambio de valor del input
        const input=wrapper.find('input');
        let value="Hamtaro"
        input.simulate('change', {
            target:{
            value:value
            }});
        //simulación del submit
        const form = wrapper.find("form");
        form.simulate('submit', {preventDefault(){}});
        expect(setCategory).toHaveBeenCalled;
        expect(input.props().value).toBe("");
       // expect(setCategory).toHaveBeenCalledWith( expect.any( Function ) );
    })
    
})