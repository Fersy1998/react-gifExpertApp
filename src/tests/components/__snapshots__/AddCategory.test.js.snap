// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory.js should show perfectly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Agregar CategorÃ­a"
    value=""
  />
</form>
`;
